File: react-java/backend/src/main/java/com/clt/hrm/core/administration/user/dtos/SearchUserDto.java
package com.clt.hrm.core.administration.user.dtos;

import com.clt.hrm.infra.common.dtos.SearchBaseDto;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper=false)
public class SearchUserDto extends SearchBaseDto {
    private String usrId;
    private String usrNm;
}

File: nestjs/backend/src/apps/administration/users/dto/search-user.dto.ts
import { IsOptional, IsString } from 'class-validator';
import { SearchBaseDto } from '@infra/common/dto';

export class SearchUserDto extends SearchBaseDto {
  @IsOptional()
  @IsString()
  usrId?: string;

  @IsOptional()
  @IsString()
  usrNm?: string;
}
